coalib/results/Diff.py

```diff
@@ -166,16 +166,21 @@ def from_clang_fixit(cls, fixit, file):
         """
         assert isinstance(file, (list, tuple))
 
-        oldvalue = '\n'.join(file[fixit.range.start.line-1:
-                                  fixit.range.end.line])
-        endindex = fixit.range.end.column - len(file[fixit.range.end.line-1])-1
-
-        newvalue = (oldvalue[:fixit.range.start.column-1] +
-                    fixit.value +
-                    oldvalue[endindex:])
-        new_file = (file[:fixit.range.start.line-1] +
-                    type(file)(newvalue.splitlines(True)) +
-                    file[fixit.range.end.line:])
+        try:
+            oldvalue = '\n'.join(file[fixit.range.start.line-1:
+                                      fixit.range.end.line])
+            endindex = fixit.range.end.column - \
+                len(file[fixit.range.end.line-1])-1
+
+            newvalue = (oldvalue[:fixit.range.start.column-1] +
+                        fixit.value +
+                        oldvalue[endindex:])
+            new_file = (file[:fixit.range.start.line-1] +
+                        type(file)(newvalue.splitlines(True)) +
+                        file[fixit.range.end.line:])
+        except LibclangError as err:
+            logging.warning('coala requires clang 3.4')
+            raise err
```

Comment:
I totally agree with you suggestion. To check the version  we can use
```
except LibclangError as err:
    if 'libclang.so version' in str(err):
        logging.error('coala requires clang 3.4')
        raise err 
```
to be more precise but we first need to catch the [LibclangError](https://github.com/llvm-mirror/clang/blob/a6d3ef724655a64ac137945615d3ff002c4f7993/bindings/python/clang/cindex.py#L4019) somehow so that we can know the error is due to the incompatible version.

We could have used the `__version__` attribute but it is not present in [clang.cindex](https://github.com/llvm-mirror/clang/blob/a6d3ef724655a64ac137945615d3ff002c4f7993/bindings/python/clang/cindex.py) module.